_source/_posts/2022-02-28-shopping-app-with-node-minio.md

```diff
@@ -0,0 +1,709 @@
+      useInteractionCodeFlow: true,
+      scopes: ['openid', 'email', 'profile'],
+    },
+  };
+  var oSignIn = new OktaSignIn(signInWidgetConfig);
+  oSignIn.showSignInAndRedirect({
+    el: '#okta-login-container'
+  }).catch(function(error) {
+    alert('Catch unknown errors here which widget does not support');
+  });
+</script>
+{{/if}}
+```
+{% endraw %}
+
+The flow parameter is not needed for the login flow since it's the default flow for the Sign-In Widget.
+
+{% img
+blog/shopping-app-with-node-minio/login.png
+alt:"Login page"
+width:"800" %}{: .center-image }
+
+### Register page
+
+Create a file called `register.handlebars`and add the following code to it:
+
+{% raw %}
+```html
+<h1 class="text-center">Please Register</h1>
+<div id="okta-register-container">
+</div>
+<script src="../js/okta-config.js"></script>
+<script>
+  const signInWidgetConfig = {
+    flow: 'signup',
+    baseUrl: credentials.domain,
+    useInteractionCodeFlow: true,
+    clientId: credentials.clientId,
+    redirectUri: window.location.origin + '/callback',
+    authParams: {
+      clientId: credentials.clientId,
+      pkce: true,
+    },
+  };
+  var fSignIn = new OktaSignIn(signInWidgetConfig);
+  fSignIn.showSignInAndRedirect({
+    el: '#okta-register-container'
+  }).catch(function(error) {
+    alert('Catch unknown errors here which widget does not support');
+  });
+</script>
+```
+{% endraw %}
+
+Here, we specify `flow: 'signup'` to initialize the Sign-In Widget flows to start from the register or signup screens.
+
+{% img
+blog/shopping-app-with-node-minio/register.png
+alt:"Register page"
+width:"800" %}{: .center-image }
+
+### Unlock page
+
+Create a file called `unlock.handlebars`and add the following code to it:
+
+{% raw %}
+```html
+<h1 class="text-center">Unlock Account</h1>
+<div id="okta-unlock-container">
+</div>
+<script src="../js/okta-config.js"></script>
+<script>
+  const signInWidgetConfig = {
+    flow: 'unlockAccount',
+    baseUrl: credentials.domain,
+    useInteractionCodeFlow: true,
+    clientId: credentials.clientId,
+    redirectUri: window.location.origin + '/callback',
+    authParams: {
+      clientId: credentials.clientId,
+      pkce: true,
+    },
+  };
+  var fSignIn = new OktaSignIn(signInWidgetConfig);
+  fSignIn.showSignInAndRedirect({
+    el: '#okta-unlock-container'
+  }).catch(function(error) {
+    // This function is invoked with errors the widget cannot recover from:
+    // Known errors: CONFIG_ERROR, UNSUPPORTED_BROWSER_ERROR
+    alert('Catch unknown errors here which widget does not support');
+  });
+</script>
+```
+{% endraw %}
+
+Here, we specify `flow: 'unlockAccount'` to initialize the Sign-In Widget flows to start from the unlock screens.
+
+{% img
+blog/shopping-app-with-node-minio/unlock-account.png
+alt:"Unlock page"
+width:"800" %}{: .center-image }
+
+### Forgot password page
+
+Create a file called `forgot-password.handlebars` and add the following code to it:
```

Comment:
Looks like additional settings are needed to configure this flow.

Running the application shows the error:
"At this time your password can only be reset by an administrator."

Reply:
Yes if you see this error, then the password policy for forgot-password and unlock account are not checked.